src/main/java/org/dice_research/opal/civet/metrics/AvailabilityOfLicensesMetric.java

```diff
@@ -0,0 +1,175 @@
+ * 
+ * @author Gourab Sahu
+ */
+public class AvailabilityOfLicensesMetric implements Metric {
+
+	public static boolean isValidLicenseURL(String LicenseURL) {
+		/*
+		 * new URL tries to create a new URL with provided license URL. If invalid
+		 * License URL then catch exception and return false.
+		 */
+		try {
+			new URL(LicenseURL).toURI();
+			return true;
+		}
+
+		catch (Exception e) {
+			return false;
+		}
+	}
+
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final String DESCRIPTION = 
+			 "If dataset has rights/license info then award 5 stars "
+			+ "Else If dataset has no rights/license but all distributions in dataset have licenses/rights then give 5 star"
+			+ "Else If dataset has no rights/license but less than 100% and more than or equal to 75% dataset's distribution has then 4 stars are awarded"
+			+ "Else If dataset has no rights/license but less than 75% and more than or equal to 50% dataset's distribution has then 3 stars are awarded"
+			+ "Else If dataset has no rights/license but less than 50% and more than or equal to 25% dataset's distribution has then 2 stars are awarded"
+			+ "Else If dataset has no rights/license but less than 25% and more than  0% dataset's distribution has then 1 star is awarded"
+			+ "Else if no License at all then 0 star is awarded.";
+
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+
+		LOGGER.info("Processing dataset " + datasetUri);
+
+		Resource dataset = ResourceFactory.createResource(datasetUri);
+
+		// NodeIterator nodeIterator = model.listObjectsOfProperty(dataset,
+		// DCAT.keyword);
+
+		// For Score calculation
+		int TotalDistributions = 0;
+		int TotalDistributionsWithLicense = 0;
+		int TotalDistributionsWithRights = 0;
+
+		// score to return
+		int score = 0;
+
+		/*
+		 * For dcat:dataset, we will first check if license is given or not. If not
+		 * check anything with 'right'.
+		 */
+		int NumberOfLicensesInDataset = 0;
+		NodeIterator LicensesIterator = model.listObjectsOfProperty(dataset,DCTerms.license);
+		NodeIterator RightsIterator = model.listObjectsOfProperty(dataset,DCTerms.rights);
+		if (LicensesIterator.hasNext()) {
+           RDFNode licensObject = LicensesIterator.nextNode();
+		   if(!(licensObject.toString().isEmpty()) && isValidLicenseURL(licensObject.toString()))
+			   NumberOfLicensesInDataset++;   
+		}
+		else if (RightsIterator.hasNext()) {
+	           RDFNode RightsObject = RightsIterator.nextNode();
+			   if(!(RightsObject.toString().isEmpty()) && isValidLicenseURL(RightsObject.toString()))
+				   NumberOfLicensesInDataset++;   
+			}
+		
+		
+		if (NumberOfLicensesInDataset == 0) {
+			// This means there is no license/rights info in dct:dataset, we need to check
+			// each dct:distributions
+
+			NodeIterator DistributionsIteratorLicense = model.listObjectsOfProperty(dataset,DCAT.distribution);
+
+			if (DistributionsIteratorLicense.hasNext()) {
```

Comment:
no need of If statement as there is while(DistributionsIteratorLicense.hasNext()) also .

Reply:
In the while loop, I am checking for distribution resource. In the If statement, I am checking if those resource has the "rights" property or not. I am sorry, I can not remove that.

Reply:
if (DistributionsIteratorLicense.hasNext()) 
{
while (DistributionsIteratorLicense.hasNext())
{
// code
}
}

I was trying to say, there is if and while both, so if you delete If statement, it would be fine,coz the if statement is checked with the while statement also.

Reply:
Ah great. Now I understand 👍 
Yes removed :)